app/models/api/trainee_attributes/v01.rb

```diff
@@ -85,13 +96,17 @@ class V01
 
       def initialize(new_attributes = {})
         new_attributes = new_attributes.to_h.with_indifferent_access
-        super(new_attributes.slice(*TraineeAttributes::V01::ATTRIBUTES + [:nationalities]).except(
-          :placements_attributes,
-          :degrees_attributes,
-          :nationalisations_attributes,
-          :hesa_trainee_detail_attributes,
-          :trainee_disabilities_attributes,
-        ))
+
+        super(
+          new_attributes.slice(
+            *(ATTRIBUTES + INTERNAL_ATTRIBUTES) + %i[nationalities],
```

Comment:
maybe just merge it 

```ruby 
ATTRIBUTES = ([] + (INTERNAL_ATTRIBUTES)).freeze
```

Reply:
We don't want to have everything in the `ATTRIBUTES` constant because we will then permit the `INTERNAL_ATTRIBUTES` to be set through the [controller](https://github.com/DFE-Digital/register-trainee-teachers/blob/main/app/controllers/api/trainees_controller.rb#L71) 😉